test/gtest/uct/uct_test.h

```diff
@@ -186,6 +186,35 @@ class uct_test : public testing::TestWithParam<const resource*>,
         uct_iov_t               m_iov;
     };
 
+    class uct_tl_names {
```

Comment:
why need nested class and member variables and add std::vector<string> overload for has_transport?
can just define has_ud(), has_rc(), has_rc_or_dc() in uct_test

Reply:
good point, it simplifies
done